buzzmobile/tests/test_utils/launch_tools.py

```diff
@@ -44,13 +51,13 @@ def launcher(func):
         def new_test(self):
             """Wrapper around the user provided test that runs a launch file.
             """
-            launch = ROSLauncher(full_name)
+            os.environ['ROS_MASTER_URI'] = self.rosmaster_uri
+            launch = ROSLauncher(full_name, port=self.port)
```

Comment:
Shouldn't we, for instance, also be checking whether `_LAUNCHER[self.port]` already exists?